app/src/main/java/com/android/unio/model/notification/CloudMessaging.kt

```diff
@@ -0,0 +1,143 @@
+package com.android.unio.model.notification
+
+import android.app.Notification
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import com.android.unio.R
+import com.google.firebase.Firebase
+import com.google.firebase.functions.functions
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+/**
+ * Enum class representing the different types of notifications that can be broadcasted.
+ *
+ * @param requiredFields The fields that are required for the notification type.
+ * @param displayName The name of the notification type. This will be shown to the user.
+ * @param description The description of the notification type.
+ */
+enum class NotificationType(
+    val requiredFields: List<String>,
+    val displayName: String,
+    val description: String
+) {
+  EVENT_SAVERS(
```

Comment:
This might seem a bit redundant but I believe the required fields should be put in an enum somewhere. It'll be easier to check with a for each and a bit more readable an modulable... But it is weird to have an enum in an enum, so this is a minor change